var/spack/repos/builtin/packages/parsec/package.py

```diff
@@ -53,3 +56,28 @@ def cmake_args(self):
             self.define_from_variant('PARSEC_DEBUG_PARANOID', 'debug_verbose'),
         ]
         return args
+
+    def check(self):
```

Comment:
```suggestion
    @run_after('install')
    def check(self):
```

Reply:
This change causes the check method to run all the time after install, even when `--test root` is not passed as an argument to `spack install`.
These are quite a lot of tests, maybe they should not run all the time. I am misunderstanding something about how `--test` works?

Reply:
You are correct.  You would need to add the following between the `run_after` and `def check` lines if you only want the tests to be run when the `--test` option is specified on the command line:

```suggestion
    @on_package_attributes(run_tests=True)
    def check(self):
```